src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -984,36 +922,45 @@ public float CalculateSpeed()
 
     public float CalculateHitpoints()
     {
-        var maxHitpoints = Membrane.Hitpoints +
-            (Rigidity * Constants.MEMBRANE_RIGIDITY_HITPOINTS_MODIFIER);
+        var maxHitpoints = CurrentSpecies.MembraneType.Hitpoints +
+            (CurrentSpecies.MembraneRigidity * Constants.MEMBRANE_RIGIDITY_HITPOINTS_MODIFIER);
 
         return maxHitpoints;
     }
 
     /// <summary>
     ///   Returns the cost of the organelle that is about to be placed
     /// </summary>
-    public float CalculateCurrentOrganelleCost()
+    public int MutationPointsAfterOrganelleAddition()
     {
+        if (MovingOrganelle != null)
+        {
+            return Constants.BASE_MUTATION_POINTS - Constants.ORGANELLE_MOVE_COST;
+        }
+
         if (string.IsNullOrEmpty(ActiveActionName) || !ShowHover)
-            return 0;
+            return Constants.BASE_MUTATION_POINTS;
 
-        var cost = SimulationParameters.Instance.GetOrganelleType(ActiveActionName).MPCost;
+        var prospectiveSpecies = (MicrobeSpecies)CurrentSpecies.Clone();
```

Comment:
I don't have an immediately good design to suggest here, but I dislike just accumulating temporary objects for the garbage collector to deal with in this new design.